src/mechanics/cell/Jacobian.h

```diff
@@ -21,53 +21,71 @@ class Jacobian
             switch (globalDimension)
             {
             case 1:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<1>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 1, 1> jacobian = CalculateFixedSize<1,1>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 2:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<2>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 2, 2> jacobian = CalculateFixedSize<2,2>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 3:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<3>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 3, 3> jacobian = CalculateFixedSize<3,3>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
             }
+            }
         }
-        else
+        else if (globalDimension == interpolationDimension + 1)
         {
-            assert(globalDimension == interpolationDimension + 1);
             switch (globalDimension)
             {
             case 2:
             {
-                const int numRows = derivativeShapeFunctions.rows();
-                NodeValues nodeCopy = nodeValues;
-                // see CalculateFixedSize(...) for the Eigen::Map functionality
-                auto nodeBlockCoordinates =
-                        Eigen::Map<Eigen::Matrix<double, 2, Eigen::Dynamic>>(nodeCopy.data(), 2, numRows);
-                mDetJacobian = (nodeBlockCoordinates * derivativeShapeFunctions).norm();
+                Eigen::Matrix<double, 2, 1> jacobian = CalculateFixedSize<2,1>(nodeValues, derivativeShapeFunctions);
+                Eigen::Matrix<double, 2,2> extendedJacobian;
+                extendedJacobian.col(0) = jacobian;
```

Comment:
IMO you can assign the result of `CalculateFixedSize<2,1>` directly to `extendedJacobian.col(0)` without this temporary `jacobian` variable.